common/src/program_storage.rs

```diff
@@ -37,13 +37,23 @@ pub trait Error {
     fn cannot_find_page_data() -> Self;
 }
 
+/// The entity specifies the structure of stored data.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, TypeInfo)]
+#[codec(crate = codec)]
+#[scale_info(crate = scale_info)]
+pub struct Item<BlockNumber> {
+    pub program: Program,
+    pub block_number: BlockNumber,
+    pub interval: BlockNumber,
```

Comment:
I would suggest naming this as `period`. Interval is more like a range with start and end. You even use the `period` term in other places of the code. It would also be useful to have some comment about what this period means, or maybe it can be reflected in the name somehow like `hold_period`